src/lib/RequestParser.js

```diff
@@ -139,7 +142,21 @@ class RequestParser { // eslint-disable-line no-unused-vars
         }
 
         const flags = object.flags || Nimiq.Transaction.Flag.NONE;
-        const data = object.data || new Uint8Array(0);
+
+        const data = object.data ? this.parseMessage(object.data) : new Uint8Array(0);
+        if (flags === Nimiq.Transaction.Flag.NONE && data.length > 64) {
+            throw new Errors.InvalidRequestError('Data must not exceed 64 bytes');
+        }
+        if (flags === Nimiq.Transaction.Flag.CONTRACT_CREATION
+                && data.length !== 78 // HTLC
+                && data.length !== 24 // Vesting
+                && data.length !== 36 // Vesting
+                && data.length !== 44) { // Vesting
+            throw new Errors.InvalidRequestError(
+                'Contract creation data must equal 78 bytes for HTLC and 24, 36, or 44 bytes for vesting contracts',
```

Comment:
I think this error message can be simplified to `Invalid data length`.

Reply:
It looses information if I do. I don't think simplification is necessarily good, if it is acieved by removing useful information.

Reply:
Ok, but then it is still _technically_ incorrectd, because you write that the _data_ must equal that, but what you mean is the data _length_.

Reply:
I'd add 'length' then, rather than shortening it.

Reply:
```suggestion
                'Contract creation data must be 78 bytes for HTLC and 24, 36, or 44 bytes for vesting contracts',
```